packages/std/src/imports.rs

```diff
@@ -367,6 +384,151 @@ impl Api for ExternalApi {
         Ok(Addr::unchecked(address))
     }
 
+    #[cfg(feature = "cosmwasm_2_1")]
+    fn bls12_381_aggregate_g1(&self, g1s: &[u8]) -> Result<[u8; 48], VerificationError> {
+        let point = [0_u8; 48];
+
+        let send = build_region(g1s);
+        let send_ptr = &*send as *const Region as u32;
+
+        let out = build_region(&point[..]);
```

Comment:
Does it? Because I followed the pattern as done in, for example, `secp256k1_verify`.  
It made sense to me to construct it directly from the slice since we call out to the host and they can then just index the linear WASM memory.  

The assumption that the region is heap allocated only comes into play when we `consume_region`, which we don't do, no?

Reply:
The entire region concept originally assumes to operate in the linear memory of Wasm, where we have offsets from the beginning of the memory and no pointers:

```
/// Describes some data allocated in Wasm's linear memory.
/// A pointer to an instance of this can be returned over FFI boundaries.
///
/// This struct is crate internal since the cosmwasm-vm defines the same type independently.
#[repr(C)]
pub struct Region {
    /// The beginning of the region expressed as bytes from the beginning of the linear memory
    pub offset: u32,
```

Now if you provide a stack variable, I would assume the host cannot even write to it. Maybe it works by accident if Wasm stores the stack variable to the linear memory too. But I would not count on that.

What we do in `secp256k1_verify` is indeed questionable since we don't know where those references are coming from.

Reply:
The new region system landed in main, so I rebased this branch. Based on the discussions we had when designing it, this should be fine.  
So I'm gonna mark the conversations as resolved for now.